onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,102 @@
+    """
+    user = User.objects.get(username=username)
+    # Get the number of projects for this user
+    user_projects = Project.objects.filter(created_by=user).count()
+    # Get the number of forms
+    user_forms = XForm.objects.filter(user=user).count()
+    # Get the number of submissions
+    user_sumbissions = Instance.objects.filter(user=user).count()
+
+    result = {
+        'projects': user_projects,
+        'forms': user_forms,
+        'submissions': user_sumbissions
+    }
+
+    return result
+
+
+def inactivate_user(username):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write(
+            sys.style.SUCCESS(
+                'User {} deleted successfully.'.format(username)))
+        # confirm too that no user exists with provided email
+        if len(User.objects.filter(email=user.email, is_active=True)) > 1:
+            inactivate_user(user.username)
+    except User.DoesNotExist:
+        sys.stdout.write('User {} does not exist.'.format(username))
+
+
+class Command(BaseCommand):
+    """
+    Delete users management command.
+    """
+    help = 'Delete users'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--user-details', nargs='*')
+
+        parser.add_argument(
+            '--user_input',
+            help='Confirm deletion of user account',
+            default=False
+        )
+
+    def handle(self, *args, **kwargs):
+        users = kwargs.get('user')
+        user_input = kwargs.get('user_input')
+
+        if users:
+            for user in users:
+                username, email = user.split(':')
+                user_stats = get_user_object_stats(username)
+                if user_input is False:
+                    # If the --user_input flag is not provided.
+                    # Get acknowledgement from the user on this
+                    user_response = input(
+                        "User account '{}' has {} projects, {} forms and {} submissions. \
+                        Do you wish to continue deleting this account?".format(
+                            username,
+                            user_stats['projects'],
+                            user_stats['forms'],
+                            user_stats['submissions']
+                            ))
+                    if user_response is True:
+                        inactivate_user(username)
+                        sys.stdout.write(
+                            "User Account {} deleted successfully!".format(
+                                username))
```

Comment:
`inactivate_user` already does this, I believe.

Reply:
https://github.com/onaio/onadata/pull/1717/files#diff-676e1a7a28413759b136ce69c33bab13R49-R51